### PR TITLE
Orv2 863

### PR DESCRIPTION
**Pull request by bcgov-brwang**
# Description
These changes is base on the requirement for editing applications.
[https://xd.adobe.com/view/5b1bae26-6f0d-4870-896c-8efdc5854158-0e48/screen/b2c11a91-5193-42e7-a685-c8bd5581f3a0/](url)
User can be navigated to the edit view by clicking the hyper link on the application in progress list.

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Test steps:

1. Login to frontend home page.
2. Click "Permits" tab.
3. Create a new permit application.
4. Click "Applications" tab.
5. Find the new created in the previous step.
6. Click on the hyper link of the application.
7. Expect to be navigated to edit page.
8. Inspect the Edit page, verify if application number, created date and updated date are rendered correctly.
9. Update the fields of the application, and click "Save".
10. The "Last Updated" field should display the updated value.
11. Click "Leave" button.
12. Verify if a leave confirmation dialog can be displayed and behave correctly.

- [X] Local tested

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend - Vehicles](https://onroutebc-413-backend-vehicles.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://onroutebc-413-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/features/manageVehicles/components/form/PowerUnitForm.tsx
Comment (devin-aot) [inline frontend/src/features/manageVehicles/components/form/PowerUnitForm.tsx:1]: These fields need to be sent as a number to the backend to satisfy the backend validations. That's why valueAsNumber is set to true. I understand that you probably saw the console warning/error, but we need to find a solution where we still send it as a number type.

File: frontend/src/features/manageVehicles/components/form/TrailerForm.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export const TrailerForm = ({ trailer }: TrailerFormProps) => {
               name: "year",
               rules: {
                 required: { value: true, message: "Year is required." },
-                valueAsNumber: true,
+                // valueAsNumber: true,
```
Comment (devin-aot) [inline frontend/src/features/manageVehicles/components/form/TrailerForm.tsx]: Same comment as above. Needs to be sent to backend as a number, not a string

File: frontend/src/features/permits/pages/TermOversize/form/VehicleDetails/VehicleDetails.tsx [hunk 2]
```diff
@@ -195,7 +195,7 @@ export const VehicleDetails = ({ feature }: { feature: string }) => {
             name: "permitData.vehicleDetails.year",
             rules: {
               required: { value: true, message: "Year is required." },
-              valueAsNumber: true,
+              // valueAsNumber: true,
```
Comment (devin-aot) [inline frontend/src/features/permits/pages/TermOversize/form/VehicleDetails/VehicleDetails.tsx]: Needs to be a number

File: frontend/src/features/permits/types/application.d.ts [hunk 3]
```diff
@@ -59,9 +59,9 @@ export interface Commodities {
 }
 
 export interface TermOversizeApplication {
-  startDate: Dayjs;
+  startDate: Dayjs | any;
```
Comment (devin-aot) [inline frontend/src/features/permits/types/application.d.ts:62]: Could be a good time to setup a meeting with the devs to discuss how we want to handle date formats and how we send them between FE/BE/DB

**Review by gchauhan-aot — COMMENTED** [review]
Backend changes look fine

**Review by devin-aot — APPROVED** [review]

**Pull request merged**